discretenet/problems/gisp/gisp.py

```diff
@@ -0,0 +1,216 @@
+from pathlib import Path
+import random
+from typing import Union
+import pyomo.environ as pyo
+import networkx as nx
+try:
+    from importlib.resources import open_text
+except ImportError:
+    from importlib_resources import open_text
+
+from discretenet.problem import Problem
+from discretenet.generator import Generator
+
+
+class GISPProblem(Problem):
```

Comment:
A concern with the way that the problem and generator are implemented here is that it would be difficult to re-instantiate a problem without the generator, since the graph and such is created in the generator.

I added an abstract `get_parameters()` method to the base `Problem` class, which should return the parameters used to instantiate the problem (you'll need to add it to this concrete class). I had intended for that to be saved as a JSON file when the problem is saved and we would publish that, so that anyone could re-instantiate a problem with just those inputs. 

At the moment, that won't be possible because of some of that logic being done in the generator instead. Do you think it would be feasible to move a lot of the generator logic from the `generate()` method like `generate_rev_costs()` into the problem class instead, or would that end up with a bunch of duplicated computation?

I think the networkx graph is going to be the biggest trouble here. The only way that I can think of being able to pass it in as a JSON-serializable parameter to the problem would be passing a string to a filename, but then every single instance would need to read that file. I'll give that some thought and maybe switch to pickle instead of json.